lib/phantom_script.js

```diff
@@ -42,45 +38,56 @@ var renderPage = function(url) {
     });
   };
 
-  var dumpPageContent = function () {
+  var dumpPageContent = function(){
     var output = page.content;
     output     = output.replace(/<script[^>]+>(.|\n|\r)*?<\/script\s*>/ig, '');
     output     = output.replace('<meta name="fragment" content="!">', '');
+    var rem    = /<!--([\ ]{0,2})response:status-code=([0-9]{3})([\ ]{0,2})-->/.exec(output);
 
-    if(output.indexOf('<!-- response:status-code=404 -->') > -1) {
-      status = 404;
+    if(rem && rem.length >= 3){
+      if(!isNaN(rem[2])){
+        realStatus = parseInt(rem[2]);
+      }
     }
 
     console.log(JSON.stringify({
-      status: status,
+      status:  realStatus,
       headers: headers,
       content: output
     }));
   };
 
-  page.onResourceReceived = function(response) {
+  page.onResourceReceived = function(response){
+    if(response.url && response.url.length){
+      response.url = response.url.replace(/\/$/, "");
+    }
+
+    if(response.redirectURL && response.redirectURL.length){
+      response.redirectURL = response.redirectURL.replace(/\/$/, "");
+    }
+
     if(response.url === url || response.redirectURL === url){
-      status  = response.status;
-      headers = response.headers;
+      realStatus  = response.status;
+      headers     = response.headers;
     }
   };
 
-  page.open(url, function(status) {
-    if (status === 'fail'){
+  page.open(url, function(status){
+    if(status === 'fail'){
       phantom.exit();
     }
     var renderIterations = 0;
-    intervalId = setInterval(function() {
+    intervalId = setInterval(function(){
       var renderStatus = isReady();
-      if(renderIterations < 50 && (!renderStatus || status === null || status == 'null')){
+      if(renderIterations < 50 && (!renderStatus || realStatus === null || realStatus == 'null')){
```

Comment:
Not sure, it was added few releases ago in a process of testing and debugging. I guess there is no reason to get `null` (string), but I really doesn't remember why we did it like this.